### PR TITLE
Add cov_flash()

### PR DESCRIPTION
**Pull request by gaow**
@pcarbo we would like to add this `cov_flash` function to formalize the data driven prior pipeline with FLASH. I'll comment in the code for what we need your input for.

### CODE REVIEW

File: R/data2cov.R [hunk 1]
```diff
@@ -32,6 +32,78 @@ cov_pca = function(data,npc,subset = NULL){
...
+  else flash_factors = as.matrix(f$ldf$f)
+  if (!is.null(output_model)) saveRDS(list(model=f, factors=flash_factors), output_model)
+  if (missing(tag)) tag = args$init_fn
+  U.flash = list()
+  U.flash[[paste0("tFLASH_", tag)]] = t(f$fitted_values) %*% f$fitted_values / nrow(f$fitted_values)
+  if (ncol(flash_factors)>0) U.flash = c(U.flash, c(cov_from_factors(t(flash_factors), paste0("FLASH_", tag))))
+  return(U.flash)
+}
+
+nonneg <- function(Y, K = 1) {
```
Comment (gaow) [inline R/data2cov.R:96]: @pcarbo we have to use `flashr:::udv_si` here which does not look very nice. What do you suggest? Do you think we should implement instead in flashr the non-negative factor constraint?

Comment (pcarbo) [reply R/data2cov.R:96]: @gaow `udv_si` is a very simple function that just calls `softImpute`. Why not just call `softImpute` directly?

Comment (gaow) [reply R/data2cov.R:99]: Thanks @pcarbo done. But with this patch we'll have to make flashr a dependency. Is that okay?

Comment (pcarbo) [reply R/data2cov.R:99]: I think that is fine.

**Pull request merged**